src/impl/apis/data_api_impl.py

```diff
@@ -258,35 +259,152 @@ async def get_time_range(
     return time_range
 
 
+async def get_query_attributes(query_parameters, token_model, pool):
```

Comment:
Types, and the name isn't ideal. Something like `get_table_names` would be a bit clearer?